src/common/components/ui/PreviewBadge.tsx

```diff
@@ -53,14 +55,15 @@ const PreviewBadge: React.FC<PreviewBadgeProps> = ({
         onClick={(e) => e.preventDefault()}
         icon={<InfoCircleIcon color="var(--pf-c-label__content--Color)" />}
         className={clsName}
+        data-testid={`${idPrefix}-support-level`}
       >
         {text}
       </Label>
     </Popover>
   );
 };
 
-export type DeveloperPreviewProps = Pick<PreviewBadgeProps, 'className' | 'position'>;
+export type DeveloperPreviewProps = Pick<PreviewBadgeProps, 'className' | 'position' | 'idPrefix'>;
```

Comment:
We already have a type called `WithTestID`, which you can use in order to extend the props of some component with a `testId` member ([example](https://github.com/openshift-assisted/assisted-ui-lib/blob/07da563e63701afbeae15ac430d1ca33a85c86b2/src/common/components/ui/DetailList.tsx#L47))

Reply:
This doesn't get a test id it gets an id prefix the testId itself is generated in the PreviewBadge component

Reply:
I still don't understand why the badges need this prop? is it for automation? if it is then you should use a different name for that prop (maybe `testId`) since `idPrefix` is used in other places in order to put id attributes on the underlying components ([example](https://github.com/openshift-assisted/assisted-ui-lib/blob/5e6951917760e4f5c40049806302c1b45110943b/src/ocm/components/AddHosts/InventoryAddHost.tsx#L30)).